api/pkg/apis/v1alpha1/vendors/solution-vendor.go

```diff
@@ -245,14 +260,25 @@ func (c *SolutionVendor) onReconcile(request v1alpha2.COARequest) v1alpha2.COARe
 				Body:  []byte(err.Error()),
 			})
 		}
-		delete := request.Parameters["delete"]
+		isRemove := request.Parameters["delete"]
 		targetName := ""
 		if request.Metadata != nil {
 			if v, ok := request.Metadata["active-target"]; ok {
 				targetName = v
 			}
 		}
-		summary, err := c.SolutionManager.Reconcile(ctx, deployment, delete == "true", namespace, targetName)
+
+		instance := deployment.Instance.ObjectMeta.Name
+		sLog.InfofCtx(ctx, "V (Solution): onReconcile create context with timeout, instance: %s, job id: %s, isRemove: %s", instance, deployment.JobID, isRemove)
```

Comment:
I would prefer you move saving previous cancel func and house keeping job to Solution.Reconcile, vendor should be very simple.